lib/matplotlib/tests/test_figure.py

```diff
@@ -709,6 +709,86 @@ def test_removed_axis():
     fig.canvas.draw()
 
 
+def test_figure_clear():
+    # we test the following figure clearing scenarios:
+    fig = plt.figure()
+
+    # a) an empty figure
+    fig.clear()
+    assert fig.axes == []
+
+    # b) a figure with a single unnested axes
+    ax = fig.add_subplot(111)
+    fig.clear()
+    assert fig.axes == []
+
+    # c) a figure multiple unnested axes
+    axes = [fig.add_subplot(2, 1, i+1) for i in range(2)]
+    fig.clear()
+    assert fig.axes == []
+
+    # d) a figure with a subfigure
+    gs = fig.add_gridspec(ncols=2, nrows=1)
+    subfig = fig.add_subfigure(gs[0])
+    subaxes = subfig.add_subplot(111)
+    fig.clear()
+    assert subfig not in fig.subfigs
+    assert fig.axes == []
+
+    # e) a figure with a subfigure and a subplot
+    subfig = fig.add_subfigure(gs[0])
+    subaxes = subfig.add_subplot(111)
+    mainaxes = fig.add_subplot(gs[1])
+
+    # e.1) removing just the axis leaves the subplot
+    mainaxes.remove()
+    assert [subaxes] == fig.axes
+
+    # e.2) removing just the subaxis leaves the subplot
+    # and subfigure
+    mainaxes = fig.add_subplot(gs[1])
+    subaxes.remove()
+    assert [mainaxes] == fig.axes
+    assert subfig in fig.subfigs
+
+    # e.3) clearing the subfigure leaves the subplot
+    subaxes = subfig.add_subplot(111)
+    assert all(ax in fig.axes for ax in [mainaxes, subaxes])
```

Comment:
There are only two; I think we can just write out both lines, since that corresponds with what happens after `clear`.
```suggestion
    assert mainaxes in fig.axes
    assert subaxes in fig.axes
```